src/main/java/com/minecolonies/structures/client/TemplateTessellator.java

```diff
@@ -0,0 +1,147 @@
+    {
+        if (!isReadOnly)
+        {
+            this.builder.finishDrawing();
+            this.vboUploader.draw(this.builder);
+            this.isReadOnly = true;
+        }
+
+        preTemplateBufferBinding(rotation, mirror, drawingOffset, inTemplateOffset);
+
+        this.buffer.bindBuffer();
+
+        preTemplateDraw();
+
+        this.buffer.drawArrays(7);
+
+        postTemplateDraw();
+
+        this.buffer.unbindBuffer();
+
+        postTemplateBufferUnbinding();
+    }
+
+    private static void preTemplateBufferBinding(final Rotation rotation, final Mirror mirror, final Vector3d drawingOffset, final BlockPos inTemplateOffset)
+    {
+        final ITextureObject textureObject = Minecraft.getMinecraft().getTextureMapBlocks();
+        GlStateManager.bindTexture(textureObject.getGlTextureId());
+
+        GlStateManager.pushMatrix();
+        GlStateManager.translate(drawingOffset.x, drawingOffset.y, drawingOffset.z);
+
+        BlockPos rotateInTemplateOffset = inTemplateOffset.rotate(rotation);
+        GlStateManager.translate(-rotateInTemplateOffset.getX(), -rotateInTemplateOffset.getY(), -rotateInTemplateOffset.getZ());
+
+        RenderUtil.applyRotationToYAxis(rotation);
+        RenderUtil.applyMirror(mirror, inTemplateOffset);
+
+        GlStateManager.scale(0.995f, 0.995f, 0.995f);
+        GlStateManager.enableBlend();
+        GlStateManager.blendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+        GlStateManager.color(1F, 1F, 1F, 1F);
+        GlStateManager.pushMatrix();
+    }
+
+    private static void preTemplateDraw()
+    {
+        GlStateManager.glEnableClientState(32884);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.defaultTexUnit);
+        GlStateManager.glEnableClientState(32888);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.lightmapTexUnit);
+        GlStateManager.glEnableClientState(32888);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.defaultTexUnit);
+        GlStateManager.glEnableClientState(32886);
+
+        GlStateManager.glVertexPointer(3, 5126, 28, 0);
+        GlStateManager.glColorPointer(4, 5121, 28, 12);
+        GlStateManager.glTexCoordPointer(2, 5126, 28, 16);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.lightmapTexUnit);
+        GlStateManager.glTexCoordPointer(2, 5122, 28, 24);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.defaultTexUnit);
+
+        GlStateManager.disableCull();
+    }
+
+    private void postTemplateDraw()
+    {
+        GlStateManager.enableCull();
+
+        for (VertexFormatElement vertexformatelement : DefaultVertexFormats.BLOCK.getElements())
+        {
+            VertexFormatElement.EnumUsage vertexformatelement$enumusage = vertexformatelement.getUsage();
+            int k1 = vertexformatelement.getIndex();
+
+            switch (vertexformatelement$enumusage)
+            {
+                case POSITION:
+                    GlStateManager.glDisableClientState(32884);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 32884 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)